ui/src/app/utils/stores.tsx

```diff
@@ -0,0 +1,23 @@
+import {useEffect, useState} from 'react';
+import {atom, Atom} from './subscribable';
+
+interface RouteDataStore {
+  title?: string;
+}
+
+export const routeDataStore = atom <RouteDataStore>({});
+
+export interface AuthStore {
+  authLoaded: boolean;
+  isSignedIn: boolean;
+}
+
+export const authStore = atom<AuthStore>({authLoaded: false, isSignedIn: false});
+
+export function useStore<T>(theStore: Atom<T>) {
+  const [value, setValue] = useState(theStore.get());
```

Comment:
I think it makes sense and I like the symmetry of `useStore`, but seeing `useState` in a helper function is hurting by brain a bit. We can discuss this offline if it's faster, but:

Do you happen to know the mechanism by which React associates this global call to a component? Is it just tracking a stack of render calls - and binds any `useState` call to the current component in that stack? Or setting aside the implementation details, is that the right mental model?

I would probably add a method doc to this one to explain the intended usage, i.e. that it should only be called from a react function component (I believe)

Reply:
On the `useState` inside of `useStore` - I believe that is the standard pattern if you want your custom hook to run the render method of your component. Calling hooks from within hooks is an acceptable practice according to the [rules of hooks](https://reactjs.org/docs/hooks-rules.html)(just to put your mind at ease). With some more exposure to hooks I think this will become clearer.

I haven't looked at the implementation of `useState`, but you suggested sounds correct and is the right mental model. `useState` is a functional approach to `setState`

I can add some docs to this. Ideally, any custom hook functions should be prefixed with "use" and there will be no functions in the codebase that are named use_Something_ that are not react hooks.  I thought about putting this in a separate file called "hooks" - we could then put all of our custom hooks in there. Would that be better/clearer than adding comments? I can still add comments describing what the hook does and the file could have a top level comment about how to use the hooks.